### ISSUE TITLE
Eliminate Mockito dependency in Testutil

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#984]
We cannot make the methods of the `Repository` final:
  * `idClass()`,
  * `entityClass()`,
  * `entityStateType()`.

The Mockito dependency is bad by itself, but there are more problems we're having. For example, the class `AggregateMessageDispatcher` does this:
```java
    private static <I, A extends Aggregate<I, ?, ?>> AggregateRepository<I, A> mockRepository() {
        TestAggregateRepository mockedRepo = mock(TestAggregateRepository.class);
        when(mockedRepo.lifecycleOf(any())).thenCallRealMethod();
        when(mockedRepo.idClass()).thenReturn(Object.class);
        return mockedRepo;
    }
```
Returning `Object.class` for an ID is violation of our contract of identifiers.

**Comment by alexander-yevsyukov** [comment]
@armiol, FYI.

**Comment by armiol** [comment]
Resolved in #1132.
